pkg/operator/ceph/agent/agent.go

```diff
@@ -65,15 +65,15 @@ func New(clientset kubernetes.Interface) *Agent {
 }
 
 // Start the agent
-func (a *Agent) Start(namespace, agentImage, serviceAccount string) error {
-	err := a.createAgentDaemonSet(namespace, agentImage, serviceAccount)
+func (a *Agent) Start(namespace, agentImage, serviceAccount string, imagePullSecrets []v1.LocalObjectReference) error {
```

Comment:
@travisn might have other thoughts, but it seems to me that adding `imagePullSecrets` as a property of `Agent` would mean the parameter doesn't have to be passed through as many functions and our function calls aren't becoming perpetually longer in length. You added it as a property to `Operator` below, and that seems preferable to me.